### ISSUE TITLE
Emulator fails to load functions source code; deploy also not working

### ISSUE THREAD

**Issue by DaveNash** [firebase/firebase-functions#418]
Functions serve/deploy were working until recent update of firebase-tools and/or firebase-functions 

### Related issues

https://github.com/firebase/firebase-tools/issues/609

### Version info

**node:**
6.11.5
**firebase-functions:**
2.2.0
**firebase-tools:**
6.4.0
**firebase-admin:**
7.0.0

MacOS Mojave 10.14.3
### Test case

> git clone https://github.com/firebase/functions-samples.git
> cd functions-samples/authenticated-json-api/functions/
> npm install
> firebase serve --only functions --project testfunctionsproject-*****

### Steps to reproduce

> firebase serve --only functions

### Expected behavior

functions hosted locally at default port (5000)

### Actual behavior

> i  functions: Preparing to emulate functions.
> ⚠  functions: Failed to load functions source code. Ensure that you have the latest SDK by running npm i --save firebase-functions inside the functions directory.
> ⚠  functions: Error from emulator. Error occurred while parsing your function triggers.
> 
> /Users/[username]/[filename]/functions/index.js:13
exports.test = functions.https.onRequest(async (request, response) => {
>                                                        ^
> 
> SyntaxError: Unexpected token (
>     at createScript (vm.js:56:10)
>     at Object.runInThisContext (vm.js:97:10)
>     at Module._compile (module.js:542:28)
>     at Object.Module._extensions..js (module.js:579:10)
>     at Module.load (module.js:487:32)
>     at tryModuleLoad (module.js:446:12)
>     at Function.Module._load (module.js:438:3)
>     at Module.require (module.js:497:17)
>     at require (internal/module.js:20:19)
>     at /Users/[username]/.nvm/versions/node/v6.11.5/lib/node_modules/firebase-tools/lib/triggerParser.js:15:15

(This error occurs even after running npm i --save firebase-functions inside the functions directory.)

### Were you able to successfully deploy your functions?
No. 

> Error: Error occurred while parsing your function triggers.

**Comment by samtstern** [comment]
@DaveNash unfortunately this is a problem with the emulator.  Right now the emulator can only emulate Node 6 functions.  You can see from this part of the function sample you're using that it's written for Node 8:

```js
// The 'async' keyword and the use of '=>' syntax are both Node 8+
exports.test = functions.https.onRequest(async (request, response) => {
```

If you go to the `master` version of that repo you should find the Node-6 version of the function:
https://github.com/firebase/functions-samples/tree/master

And we know this situation is pretty bad, we're working on a new emulator!

**Comment by DaveNash** [comment]
Thank you @samtstern. It's weird, because I've seen the warnings, but was successfully emulating/deploying async/arrow functions for weeks.

**Comment by samtstern** [comment]
@DaveNash huh really?  I can't really explain how that worked ... 

FYI here is the relevant limitation:
https://github.com/GoogleCloudPlatform/cloud-functions-emulator/blob/master/README.md#note

As I said we know this is pretty lame and are working hard to improve it.

**Comment by DaveNash** [comment]
Is this no longer true? 
https://github.com/firebase/functions-samples/tree/master/es2017-transpile

**Comment by samtstern** [comment]
@DaveNash definitely true, if you use a transpiler you can write your code with advanced Node 8+ features and get it to run in a Node 6 environment.  I was just commenting on native Node 8 code not being compatible with the emulator.

My preferred environment is just to develop in TypeScript and then target the TypeScript compiler at Node 6, then I never have to think about this stuff and I get all the advanced TS features.

**Comment by DaveNash** [comment]
FYI, I've reverted to firebase-functions 2.1.0 and the emulator is working again. I don't have time to further investigate, but there must be (?) a transpiler included there that was dropped in 2.2.0.

**Comment by samtstern** [comment]
Huh that's super interesting thanks for noting it!

**Comment by DaveNash** [comment]
@samtstern, FYI & FWIW, I no longer think that's what did it. I restarted my machine for an unrelated reason, and found the emulator was back to not working. I deleted Functions > Node Modules, re-ran npm install to use firebase-tools ^6.0.0 and firebase-functions~2.1.0, and this time it still wasn't emulating properly. Then I switched to using Node 8.11.1 (via nvm use command) and, though the emulator warns me that I'm "using Node.js v8.11.1 but Google Cloud Functions only supports v6.11.5," the emulator runs. I subsequently re-installed latest firebase-functions (2.2.0) and -tools (7.0.0) and it's still emulating fine.
